statesman-server/src/main/java/io/appform/statesman/server/resources/HousekeepingResource.java

```diff
@@ -48,12 +56,30 @@ public Response getWorkflow(@PathParam("workflowId") String workflowId) {
     @POST
     @Timed
     @Path("/trigger/action/{actionId}")
-    @ApiOperation("trigger ction")
+    @ApiOperation("trigger Action")
     public Response triggerAction(@PathParam("action") String actionId,
                                   @Valid Workflow workflowId) {
         actionExecutor.get().execute(actionId, workflowId);
         return Response.ok()
                 .build();
     }
 
+    @POST
+    @Timed
+    @Path("/trigger/workflow/{workflowId}")
+    @ApiOperation("trigger workflow")
+    public Response triggerWorkflow(@PathParam("workflowId") String workflowId,
+                                    @Valid JsonNode update) {
+        Workflow wf = workflowProvider.get().getWorkflow(workflowId).orElse(null);
+        if (wf == null) {
+            return Response.status(Response.Status.NOT_FOUND)
```

Comment:
Let's not return 404's. As a protocol let's follow 204 for content not found. Let Jetty/Jersey do 404 for wrong api calls